source/validate_datarules.cpp

```diff
@@ -190,6 +190,33 @@ spv_result_t ValidateSpecConstBoolean(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Records the <id> of the forward pointer to be used for validation.
+spv_result_t ValidateForwardPointer(ValidationState_t& _,
```

Comment:
It's arguably excessive to make this a full blown "Validation" routine since no checks are occurring in this method.  You could have just directly called RegisterForwardPointer instead of calling this function.  But I can see how it's nice to maintain parallel structure at the call site.

Ok to keep.